lms/djangoapps/courseware/tests/test_inline_analytics_integration.py

```diff
@@ -35,62 +35,29 @@ def setUp(self):
         }
         json_analytics_data = json.dumps(analytics_data)
         self.data = {"data": json_analytics_data}
+        self.zendesk_response = ('A problem has occurred retrieving the data, to report the problem click '
+                                 '<a href="https://zendesk.com/hc/en-us/requests/new">here</a>')
 
-    @override_settings(ANALYTICS_ANSWER_DIST_URL='dummy_url')
-    def test_regular_user(self):
-
-        request = self.factory.post('', self.data)
-        request.user = self.user
-
-        response = get_analytics_answer_dist(request)
-        self.assertEquals(response.content, 'A problem has occurred retrieving the data, please report the problem.')
-
+    @override_settings(ZENDESK_URL='https://zendesk.com')
     def test_no_url(self):
 
         request = self.factory.post('', self.data)
         request.user = self.instructor
 
         response = get_analytics_answer_dist(request)
-        self.assertEquals(response.content, 'A problem has occurred retrieving the data, please report the problem.')
-
-    @override_settings(ANALYTICS_ANSWER_DIST_URL='dummy_url')
-    @patch('urllib2.urlopen')
-    @patch('courseware.views.process_analytics_answer_dist')
-    def test_staff_and_url(self, mock_process_analytics, mock_requests):
-
-        mock_resp = MagicMock()
-        mock_read = MagicMock(return_value="{}")
-        mock_resp.read = mock_read
-        mock_requests.return_value = mock_resp
-
-        factory = self.factory
-        request = factory.post('', self.data)
-        request.user = self.staff
-
-        mock_process_analytics.return_value = [{'dummy': 'dummy'}]
-        response = get_analytics_answer_dist(request)
-        self.assertEquals(response, [{'dummy': 'dummy'}])
+        self.assertEquals(response.content, self.zendesk_response)
 
-    @override_settings(ANALYTICS_ANSWER_DIST_URL='dummy_url')
-    @patch('urllib2.urlopen')
-    @patch('courseware.views.process_analytics_answer_dist')
-    def test_instructor_and_url(self, mock_process_analytics, mock_requests):
+    def test_no_url_no_zendesk(self):
 
-        mock_resp = MagicMock()
-        mock_read = MagicMock(return_value="{}")
-        mock_resp.read = mock_read
-        mock_requests.return_value = mock_resp
-
-        factory = self.factory
-        request = factory.post('', self.data)
+        request = self.factory.post('', self.data)
         request.user = self.instructor
 
-        mock_process_analytics.return_value = [{'dummy': 'dummy'}]
         response = get_analytics_answer_dist(request)
-        self.assertEquals(response, [{'dummy': 'dummy'}])
+        self.assertEquals(response.content, 'A problem has occurred retrieving the data.')
 
     def test_process_analytics_answer_dist(self):
 
+        # Data that is complete.
```

Comment:
As a general rule, we should never use single line comments on the first line of a unit test. At the least, we should convert it into a docstring. If a docstring is set, that message will be displayed upon test failure, which can often be very informative.

```python
def test_what(self):
    """
    Ensure that data is complete when successfully processed
    """
```

Taking it a step further, for a comment like this, I'll generally just use it to update the name of the test method (which will also be displayed upon failure if no docstring is set)

```python
def test_data_is_complete(self):
    pass
```